File: src/etch/bindings/csharp/msg/EqualsHelper.cs
/// $Id$
/// Created by Champakesan, Badri Narayanan on Jun 11, 2007.
/// Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
        
using System;

namespace Etch.Msg
{
        /// <param name="a"></param>
        /// <param name="b"></param>
        /// <returns>true if the value of a and b are equal even though the format is not 
        /// (e.g., 23 as a Byte vs. Integer).</returns>
        public delegate bool Equals(Object a, Object b);
}

File: src/etch/bindings/java/msg/EqualsHelper.java
/* $Id$
 *
 * Created by sccomer on Mar 29, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.msg;

/**
 * Interface to implementation to help compare conformant but not
 * identical objects found in struct value and array value instances.
 */
public interface EqualsHelper
{
	/**
	 * @param a
	 * @param b
	 * @return true if the value of a and b are equal even though
	 * the format is not (e.g., 23 as a Byte vs. Integer).
	 */
	public boolean equals( Object a, Object b );
}
